### PR TITLE
Integrate Persistent storage

### LINKED ISSUE #148: Integrate Persistent Storage

### PR DESCRIPTION
**Pull request by chuckluv**

### CODE REVIEW

**Review by anfee1 — APPROVED** [review]
lgtm

File: frontend/src/main/java/com/google/tripmeout/frontend/servlet/PlaceVisitParentServlet.java [hunk 1]
```diff
@@ -46,9 +49,10 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       placeService.validatePlaceId(place.placesApiPlaceId());
 
       String tripId = ServletUtil.matchUriOrThrowError(request, TRIP_NAME_PATTERN).group(1);
-
-      PlaceVisitModel newPlace =
-          place.toBuilder().setId(UUID.randomUUID().toString()).setTripId(tripId).build();
+      Key parentKey = KeyFactory.createKey(TRIP_ME_OUT_KIND_NAME, tripId);
+      String id = KeyFactory.keyToString(
+          KeyFactory.createKey(parentKey, tripId, UUID.randomUUID().toString()));
+      PlaceVisitModel newPlace = place.toBuilder().setId(id).setTripId(tripId).build();
```
Comment (KoopaKing) [inline frontend/src/main/java/com/google/tripmeout/frontend/servlet/PlaceVisitParentServlet.java:55]: The servlet shouldn't know about KeyFactory and stuff. Just use a UUID for new places. The old code was correct.

**Review by KoopaKing — CHANGES_REQUESTED** [review]

File: frontend/src/main/java/com/google/tripmeout/frontend/servlet/PlaceVisitParentServlet.java (hunk 1 above)
Comment (chuckluv) [reply frontend/src/main/java/com/google/tripmeout/frontend/servlet/PlaceVisitParentServlet.java:55]: See the issue is I need to do this in order to retrieve the placeVisit in the datastore. I couldn't figure out a different way to do this.

Comment (KoopaKing) [reply frontend/src/main/java/com/google/tripmeout/frontend/servlet/PlaceVisitParentServlet.java:55]: If you have to do this to get the datastore-based storage implementation to work, there is something wrong with the datastore-based storage.

**Comment by KoopaKing** [comment]
So #153 should unblock this integration, but most of the code here now should be tossed tbh. What you really want to do to integrate (plug in?) the persistent storage is replace the `InMemoryStorageBindingModule` in `FrontendGuiceServletContextListener` with `PersistentStorageBindingModule`.

**Pull request closed without merging**